### PR TITLE
API-40554-update-v2-526-max-request

### PR DESCRIPTION
**Pull request by stiehlrod**
## Summary

- Removes federal activation from form_526_json_api, example.json. 
- Adjusts tests that use the changed file.

## Related issue(s)

- [API-40554](https://jira.devops.va.gov/browse/API-40554)

## What areas of the site does it impact?
	modified:   modules/claims_api/config/schemas/v2/request_bodies/disability_compensation/example.json
	modified:   modules/claims_api/spec/fixtures/v2/veterans/disability_compensation/form_526_json_api.json
	modified:   modules/claims_api/spec/lib/claims_api/v2/disability_compensation_evss_mapper_spec.rb
	modified:   modules/claims_api/spec/lib/claims_api/v2/disability_compensation_pdf_mapper_spec.rb
	modified:   modules/claims_api/spec/requests/v2/veterans/526_spec.rb
	modified:   modules/claims_api/spec/requests/v2/veterans/rswag_526_spec.rb
	modified:   modules/claims_api/spec/services/disability_compensation/docker_container_service_spec.rb
	modified:   modules/claims_api/spec/services/disability_compensation/pdf_generation_service_spec.rb
	modified:   modules/claims_api/spec/sidekiq/v2/disability_compensation_docker_container_upload_spec.rb
	modified:   modules/claims_api/spec/sidekiq/v2/disability_compensation_pdf_generator_spec.rb
	modified:   modules/claims_api/app/swagger/claims_api/v2/dev/swagger.json
	modified:   modules/claims_api/app/swagger/claims_api/v2/production/swagger.json

## Acceptance criteria

- [X]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [X]  Documentation has been updated (link to documentation)
- [X]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/claims_api/app/swagger/claims_api/v2/production/swagger.json [hunk 1]
```diff
@@ -3886,7 +3876,11 @@
                                 "approximateBeginDate": "2020-06",
                                 "approximateEndDate": "2020-07"
                               }
-                            ]
+                            ],
+                            "federalActivation": {
+                              "activationDate": "2024-08-26",
+                              "anticipatedSeparationDate": "2024-10-03"
+                            }
```
Comment (tycol7) [inline modules/claims_api/app/swagger/claims_api/v2/production/swagger.json:3883]: Should we remove `federalActivation` here?

File: modules/claims_api/spec/lib/claims_api/v2/disability_compensation_pdf_mapper_spec.rb [hunk 2]
```diff
@@ -13,36 +13,36 @@
...
+      temp = JSON.parse(temp)
+      attributes = temp['data']['attributes']
+      attributes['serviceInformation']['federalActivation'] =
+        {
+          'activationDate' => activation_date,
+          'anticipatedSeparationDate' => anticipated_separation_date
+        }
+      attributes['serviceInformation']['placeOfLastOrAnticipatedSeparation'] = 'Aberdeen Proving Ground'
+      temp['data']['attributes'] = attributes
+      temp
```
Comment (tycol7) [inline modules/claims_api/spec/lib/claims_api/v2/disability_compensation_pdf_mapper_spec.rb:29]: Same question as above: do we still expect these attributes if we are removing them from the docs?

**Comment by stiehlrod** [comment]
> Maybe more of a product question, but if we are removing parameters from our docs, does that mean there is no expectation that we handle them? Or are we still supposed to handle these now-undocumented params? I am unsure why we are reinserting them in our tests after removing them from the schema - I thought the docs were meant to reflect the state of the API?

@tycol7 We have [526.json](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/claims_api/config/schemas/v2/526.json) which is the schema. The [form_526_json_api.json](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/claims_api/spec/fixtures/v2/veterans/disability_compensation/form_526_json_api.json) which is one of our example files, and the [example.json](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/claims_api/config/schemas/v2/request_bodies/disability_compensation/example.json). 

If I made changes to the schema (526.json) then I am in error. We are keeping the federal activation on the schema and therefor in the tests. Believe it or not we used to have more files and and tried to make it less convoluted with fewer files :D

File: modules/claims_api/app/swagger/claims_api/v2/production/swagger.json (hunk 1 above)
Comment (klwatanabe) [reply modules/claims_api/app/swagger/claims_api/v2/production/swagger.json:3883]: We still want to support federalActivation as a data object. And we still want to map that data to Form526. But we only expect that federalActivation object to be included for BDD claims. Whereas, our maximum request example is for a standard claim - so including it in our example adds confusion.

The side effect of including federalActivation on a standard claim request was that the zip code validation was treated like a BDD claim. For BDD claims, zip codes must be valid. So once we started mapping federalActivation data to the Form526 docker container, we were seeing errors using our example request.

Comment (FonzMP) [reply modules/claims_api/app/swagger/claims_api/v2/production/swagger.json:3883]: yeah, we're still supporting them as part of the request object, just not _that particular request example_. That being said, we don't have anything removing them if it were to _not be a BDD claim_, which is also concerning. Meaning folks can still send those dates with the request and we'll fail while submitting to the EVSS Docker Container with no noted documentation on the need to **_not provide them_** for a standard claim. 

Think we want to either add a note or some form of indicator on how we expect BDD vs Standard to receive / not receive the fedActivation dates. 

cc: @klwatanabe

Comment (FonzMP) [reply modules/claims_api/app/swagger/claims_api/v2/production/swagger.json:3883]: ignore, sounds like we're good with still sending with a BDD as long as it's valid.... and if it's submitted as invalid we're good with leaving as an errored status. so still capable of being sent in the request, and will just be validated by the docker container and report the error back if found.

Comment (tycol7) [reply modules/claims_api/app/swagger/claims_api/v2/production/swagger.json:3883]: Understood. Is there a reason why we want it in the production swagger, but not in the dev swagger?

**Comment by stiehlrod** [comment]
> > We have [526.json](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/claims_api/config/schemas/v2/526.json) which is the schema. The [form_526_json_api.json](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/claims_api/spec/fixtures/v2/veterans/disability_compensation/form_526_json_api.json) which is one of our example files we use for testing & the docs, and the [example.json](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/claims_api/config/schemas/v2/request_bodies/disability_compensation/example.json) for the docs.
> > If I made changes to the schema (526.json) then I am in error. We are keeping the federal activation on the schema and therefore in the tests. Believe it or not we used to have more files and and tried to make it less convoluted with fewer files :D
> 
> Ah, I think I understand. We still want to accept `federalActivation` and `separationLocationCode` but don't want to advertise it in the docs. So we manually add the missing params in the tests when the tests and docs diverge, since they rely on the same examples (still not sure on the difference between example.json and form_526_json_api.json, but should be ok).
> 
> These changes seem fine. Verified locally that `federalActivation` and `separationLocationCode` do not appear in the 526 v2 maximum attributes examples. One outstanding [question](https://github.com/department-of-veterans-affairs/vets-api/pull/18697#discussion_r1788025290) on why the dev and production swagger files are different, then will approve.

@tycol7 Sure, the sample.json is used for swagger. The form_526_json_api is used for testing. Search for it in our module, you will see.

File: modules/claims_api/app/swagger/claims_api/v2/production/swagger.json (hunk 1 above)
Comment (stiehlrod) [reply modules/claims_api/app/swagger/claims_api/v2/production/swagger.json:3883]: @tycol7 The diff between dev and prod swagger files is a good catch. I must have made a mistake during a merge conflict. Should be fixed now.

**Comment by tycol7** [comment]
> Sure, the sample.json is used for swagger. The form_526_json_api is used for testing.

If one file is for docs and the other is for testing, could we just remove the params from the docs file and keep the testing file intact?  So we avoid splicing the params back into the tests like so?

https://github.com/department-of-veterans-affairs/vets-api/blob/36ba3d9a2057f31a8f92c24a35d1bbe06342e2f1/modules/claims_api/spec/lib/claims_api/v2/disability_compensation_pdf_mapper_spec.rb#L17-L30

**Comment by FonzMP** [comment]
hmm.. I think there's a simplified solution here. form_526_json_api is used for testing and the literal schema definition is the 526 file. The nuance here is that the form_526_json_api file is what we use to build the maximum example request on the portal. So for example, [this block](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/claims_api/spec/requests/v2/veterans/rswag_526_spec.rb#L85-L89) sets our "maximum request/attributes" dropdown example on the dev portal.

What we _could do_ to alleviate the issue of needing to re-add it (which is I believe @tycol7's concern) is to revert the multiple specs that we've modified and manually added it back to and simply convert the line's above to some form of delete or except that would remove it in the one nuanced place.

**Comment by FonzMP** [comment]
Although the example in the form_api is of standard claim. Which means we don't require federalActivation - but it is an option to be sent, and if sent, it will be validated.

**Comment by stiehlrod** [comment]
@tycol7 @FonzMP Here is an alternative with requested feedback: https://github.com/department-of-veterans-affairs/vets-api/pull/18790/files

**Pull request closed without merging**